### DISCUSSION TITLE
Zephyr port static memory usage - what should I be expecting?

### DISCUSSION THREAD

**Discussion by gp-src** [bacnet-stack/bacnet-stack#581] [General]
I tried building the zephyr\samples\profiles\b-ss sample with Zephyr RTOS on on a nucleo_h743zi board but RAM .bss section overflows.

Build command:
`west build -b nucleo_h743zi  -p`

Loading Zephyr default modules (Zephyr base).
-- Application: C:/Users/georg/OneDrive/Documents/Projects/bacnet/bacnet_workspace/bacnet-stack/zephyr/samples/profiles/b-ss
-- CMake version: 3.27.8
-- Found Python3: C:/Users/georg/AppData/Local/Programs/Python/Python312/python.exe (found suitable exact version "3.12.0") found components: Interpreter
-- Cache files will be written to: C:/Users/georg/OneDrive/Documents/Projects/bacnet/bacnet_workspace/zephyr/.cache
-- Zephyr version: 3.4.0 (C:/Users/georg/OneDrive/Documents/Projects/bacnet/bacnet_workspace/zephyr)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
-- Board: nucleo_h743zi
-- Found host-tools: zephyr 0.16.5 (C:/Users/georg/OneDrive/Documents/Projects/zephyr/toolchain/zephyr-sdk-0.16.5)
-- Found toolchain: zephyr 0.16.5 (C:/Users/georg/OneDrive/Documents/Projects/zephyr/toolchain/zephyr-sdk-0.16.5)
-- Found Dtc: C:/ProgramData/chocolatey/bin/dtc.exe (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: C:/Users/georg/OneDrive/Documents/Projects/bacnet/bacnet_workspace/zephyr/boards/arm/nucleo_h743zi/nucleo_h743zi.dts

Linker error:
![image](https://github.com/bacnet-stack/bacnet-stack/assets/75914191/4dcc7960-47e6-4e8b-9c14-2e10f5b4dfd0)

Is this the expected memory footprint of the stack? Or am I missing something obvious to get the sample working?

**Comment by gregshue** [comment]
Great question. All the samples are not very mature yet. As I understand, the originals are demo applications for a Linux system. I don't know if they have been re-sized for MCUs yet. At this point, please assume all Zephyr integrations and Zephyr-built samples are immature.

Please note that bacnet-stack has been designed assuming integration via a leverage-and-potential-reuse paradigm. Significant work remains to evolve it for a reuse-and-extend paradigm.

**Comment by GeorgePriestner-Eaton** [comment]
I found reducing the number of supported object properties in config.h, and setting BACNET_PROTOCOL_REVISION to 14, allowed me to build my sample application OK.

**Comment by skarg** [comment]
In [Bugfix/zephyr defines kconfig](https://github.com/bacnet-stack/bacnet-stack/pull/610), I changed zephyr/CMakeLists.txt and Kconfig to use defines that are more appropriate for embedded microcontrollers.  The default src/bacnet/config.h defines controlling some RAM sizes are sized for the example apps/ that are built for full OS (win32/linux/bsd). This fixes #581.
